src/main/scala/org/allenai/plugins/VersionInjectorPlugin.scala

```diff
@@ -117,6 +140,51 @@ object VersionInjectorPlugin extends AutoPlugin {
         gitConfFile
     }
 
+  val injectCacheKeyTask = injectCacheKey := {
+    val values = ((resourceManaged in Compile).value.absolutePath, organization.value, name.value, streams.value, cacheKey.value)
+    val (resourceManagedPath, org, nameValue, s, cacheKeyResult) = values
+    val cacheKeyConfFile = new java.io.File(s"$resourceManagedPath/$org/${cleanArtifactName(nameValue)}/cacheKey.conf")
+    s.log.info(s"Generating cacheKey.conf managed resource... (cacheKey: $cacheKeyResult)")
+
+    val cacheKeyContents = s"""cacheKey: "$cacheKeyResult""""
+    IO.write(cacheKeyConfFile, cacheKeyContents)
+    cacheKeyConfFile
+  }
+
+  //Gives us the git most recent commits for all the local projects that this project depends on
+  lazy val gitMRCs = Def.taskDyn { // We have to use a dynamic task because generating tasks based on project dependencies
+    // necessarily alters the task graph
+    val MRCs = buildDependencies.value.classpathRefs(thisProjectRef.value).sorted // get the local dependencies
```

Comment:
It would be better to sort the SHA1 values, since you know that they have a stable sort order. I'm not sure what the type here is, but it looks like an sbt internal type that may change ordering on you.

Also, you probably just want to sort at the very end - the ordering in this function isn't really important . . .